src/app/api/patient/auth/signUp/route.ts

```diff
@@ -0,0 +1,28 @@
+import { getUserByEmail, patientSignUp } from "@server/mongodb/actions/User";
+import APIWrapper from "@server/utils/APIWrapper";
+import { SignupData } from "@/common_utils/types";
+
+export const POST = APIWrapper({
+  config: {
+    requireToken: true,
+  },
+  handler: async (req) => {
+    const signupData = (await req.json()) as SignupData;
+
+    const user = await getUserByEmail(signupData.email);
+    if (!user) {
+      throw new Error("User not found.");
+    }
+
```

Comment:
Could you also check if all the data in `signupData` are not undefined? Check this in one conditional statement and throw an Error saying "Missing parameter(s)."